tests/c/simple_non_serialised.c

```diff
@@ -0,0 +1,54 @@
+// Run-time:
+//   env-var: YKD_PRINT_JITSTATE=1
+//   env-var: YKD_STATS=1
+//   stderr:
+//     jit-state: start-tracing
+//     i=4
+//     jit-state: stop-tracing
+//     i=3
+//     jit-state: enter-jit-code
+//     i=2
+//     i=1
+//     jit-state: deoptimise
+//   stdout:
+//     exit
+
+// Check that basic trace compilation in a thread works.
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <yk.h>
+#include <yk_testing.h>
+
+bool test_compiled_event(YkCStats);
+
+int main(int argc, char **argv) {
+  YkMT *mt = yk_mt_new(NULL);
+  yk_mt_hot_threshold_set(mt, 0);
+  YkLocation loc = yk_location_new();
+
+  int res = 9998;
+  int i = 4;
+  NOOPT_VAL(loc);
+  NOOPT_VAL(res);
+  NOOPT_VAL(i);
+  while (i > 0) {
+    yk_mt_control_point(mt, &loc);
+    if (i == 3)
```

Comment:
Can I end up in a dead lock if I get this `i` wrong? i.e. it waits forever for the trace to compile but since I haven't reached the threshold yet that never happens? I can see myself walking into this one a lot when making tests. This is not a "can we fix this" comment, but rather a "do I need to watch out for this in the future" comment.

Reply:
Yes, if you want to wait for an event to complete, you need to make sure that you did whatever is necessary *for* that event to happen. There's no particular magic here, in that sense.